src/CMakeLists.txt

```diff
@@ -19,642 +26,297 @@ set(HYPRE_TIME    00:00:00)
 set(HYPRE_BUGS    https://github.com/hypre-space/hypre/issues)
 set(HYPRE_SRCDIR  "${PROJECT_SOURCE_DIR}")
 
+# Display the hypre version
+setup_git_version_info("${CMAKE_CURRENT_SOURCE_DIR}/../.git")
+if (GIT_VERSION_FOUND)
+  message(STATUS "Hypre Version: ${HYPRE_DEVELOP_STRING} (${HYPRE_BRANCH_NAME})")
+else()
+  message(STATUS "NOTE: Could not find .git directory")
+  message(STATUS "Hypre Version: ${HYPRE_VERSION}")
+endif()
+
+# Set the project name
 set(PROJECT_NAME HYPRE)
 project(${PROJECT_NAME}
   VERSION ${HYPRE_VERSION}
   LANGUAGES C)
 
+# Create the HYPRE library object
+add_library(${PROJECT_NAME})
+
 # We use C99 by default, but users are free to specify any newer standard version
-set(CMAKE_C_STANDARD 99)
+set(CMAKE_C_STANDARD 99 CACHE STRING "Standard to use for the C compiler")
 
+# Raise an error if the source and binary directories are the same
 if (${HYPRE_SOURCE_DIR} STREQUAL ${HYPRE_BINARY_DIR})
   message(FATAL_ERROR "In-place build not allowed! Please use a separate build directory. See the Users Manual or INSTALL file for details.")
 endif ()
 
-if (EXISTS ${HYPRE_SOURCE_DIR}/../.git)
```

Comment:
Why was this removed?  I think this has to do with the version check stuff.

Reply:
See https://github.com/hypre-space/hypre/pull/472

Reply:
Hi Rob, I moved this check to inside the new macro: https://github.com/hypre-space/hypre/blob/dd81c9716d093e3609984dbf80478747f631b988/src/config/cmake/HYPRE_CMakeUtilities.cmake#L32